source/PlayerInfo.cpp

```diff
@@ -3129,6 +3129,55 @@ void PlayerInfo::RegisterDerivedConditions()
 	visitedSystemProvider.SetGetFunction(visitedSystemFun);
 	visitedSystemProvider.SetHasFunction(visitedSystemFun);
 
+	// Read-only navigation conditions.
+	auto Days = [](const System *origin, const System *destination) -> int
+	{
+		if(!origin)
+			return -1;
+
+		auto distanceMap = DistanceMap(origin);
+		if(!distanceMap.HasRoute(destination))
+			return -1;
+		return distanceMap.Days(destination);
+	};
+
+	auto &&hyperjumpsToSystemProvider = conditions.GetProviderPrefixed("hyperjumps to system: ");
+	auto hyperjumpsToSystemFun = [this, &Days](const string &name) -> int
+	{
+		const System *system = GameData::Systems().Find(name.substr(strlen("hyperjumps to system: ")));
+		if(!system)
+		{
+			Logger::LogError("Warning: System \"" + name.substr(strlen("hyperjumps to system: "))
+					+ "\" referred to in condition is not valid.");
+			return -1;
+		}
+		return Days(this->GetSystem(), system);
```

Comment:
This is performance-wise not great, because the DistanceMap gets calculated every time the condition is evaluated.
On the other hand; we don't evaluate the conditions often, and not during realtime flight, so I guess this is okay for now.
We might need to do a follow-up PR someday to cache some data (like DistanceMaps) for expensive lookups that we do often.